src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -361,6 +361,17 @@ def _update(info, lun, value):
             _update(info_dict, lun, value)
 
 
+def is_shared_gallery_image_id(image_reference):
+    if not image_reference:
+        return False
+
+    shared_gallery_id_pattern = re.compile(r'^/SharedGalleries/[^/]*/Images/[^/]*/Versions/.*$', re.IGNORECASE)
```

Comment:
Ensure this regex covers all scenario for shared gallery image id?

Reply:
Yes, I confirmed this regex expression with @kangsun-ctrl